tests/runtime/http_server/test_http_server.py

```diff
@@ -477,41 +478,101 @@ def test_inference_streaming_sample_module(sample_task_model_id, client):
     sample_task_model_id, runtime_http_server
 ):
     """Simple check for testing a happy path unary-stream case
-    but pints the actual running server"""
+    but pings the actual running server"""
 
     for i in range(10):
         input = {"model_id": sample_task_model_id, "inputs": {"name": f"world{i}"}}
         url = f"http://localhost:{runtime_http_server.port}/api/v1/task/server-streaming-sample"
         stream = requests.post(url=url, json=input, verify=False)
         assert stream.status_code == 200
         stream_content = stream.content.decode(stream.encoding)
+        print(stream_content)
         stream_responses = json.loads(
             "[{}]".format(
-                stream_content.replace("data: ", "")
+                stream_content.replace("event: ", '{"event":')
+                .replace(
+                    StreamEventTypes.MESSAGE.value,
+                    '"' + f"{StreamEventTypes.MESSAGE.value}" + '"}',
+                )
+                .replace("data: ", "")
                 .replace("\r\n", "")
                 .replace("}{", "}, {")
             )
         )
-        assert len(stream_responses) == 10
+        assert len(stream_responses) == 20
         assert all(
             resp.get("greeting") == f"Hello world{i} stream"
             for resp in stream_responses
+            if "greeting" in resp
+        )
+        assert all(
+            resp.get("event") == StreamEventTypes.MESSAGE.value
+            for resp in stream_responses
+            if "event" in resp
+        )
+
+
+def test_inference_streaming_sample_module_actual_server_throws(
+    sample_task_model_id, runtime_http_server
+):
+    """Simple check for testing an exception in unary-stream case
+    that pings the actual running server"""
+
+    for i in range(10):
+        input = {
+            "model_id": sample_task_model_id,
+            "inputs": {"name": f"world{i}"},
+            "parameters": {"err_stream": True},
+        }
+        url = f"http://localhost:{runtime_http_server.port}/api/v1/task/server-streaming-sample"
+        stream = requests.post(url=url, json=input, verify=False)
+        assert stream.status_code == 200
+        stream_content = stream.content.decode(stream.encoding)
+        stream_responses = json.loads(
+            "[{}]".format(
+                stream_content.replace("event: ", '{"event":')
+                .replace(
+                    StreamEventTypes.ERROR.value,
+                    '"' + f"{StreamEventTypes.ERROR.value}" + '"}',
+                )
+                .replace("data: ", "")
+                .replace("\r\n", "")
+                .replace("}{", "}, {")
+            )
+        )
+        assert len(stream_responses) == 2
+        assert stream_responses[0].get("event") == StreamEventTypes.ERROR.value
+        assert (
+            stream_responses[1].get("details") == "ValueError('raising a ValueError')"
         )
+        assert stream_responses[1].get("code") == 400
```

Comment:
This is why we had to introduce catching `Value/Type` errors in the `http_server`. Previously, these errors would have been labeled witn `500`

Reply:
ah! That makes sense.